hugegraph-llm/src/hugegraph_llm/utils/embedding_utils.py

```
@@ -0,0 +1,53 @@
+from tqdm import tqdm
+
+from hugegraph_llm.models.embeddings.base import BaseEmbedding
+
+
+async def get_embeddings_parallel(embedding: BaseEmbedding, vids: list[str]) -> list[Any]:
+    sem = asyncio.Semaphore(10)
+    batch_size = 1000
+
+    # TODO: refactor the logic here (call async method)
+    async def get_embeddings_with_semaphore(vid_list: list[str]) -> Any:
+        # Executes sync embedding method in a thread pool via loop.run_in_executor, combining async programming
```

Pylint (3.3.7)

[convention] Line too long (111/100)

(C0301)